antarest/study/service.py

```diff
@@ -213,17 +228,37 @@ def _generate_timeseries(self) -> None:
                 )
             )
 
-    def run_task(self, notifier: TaskUpdateNotifier) -> TaskResult:
+    def run_task(self, notifier: TaskUpdateNotifier, listener: t.Optional[ICommandListener]) -> TaskResult:
         msg = f"Generating thermal timeseries for study '{self._study_id}'"
         notifier(msg)
-        self._generate_timeseries()
+        self._generate_timeseries(listener)
         msg = f"Successfully generated thermal timeseries for study '{self._study_id}'"
         notifier(msg)
         return TaskResult(success=True, message=msg)
 
     # Make `ThermalClusterTimeSeriesGeneratorTask` object callable
     __call__ = run_task
 
+    class TsGenerationListener(ICommandListener):
+        event_bus: IEventBus
+
+        def notify_progress(self, progress: int) -> None:
+            with db():
+                db.session.query(TaskJob).filter(TaskJob.id == self.task_id).update({TaskJob.progress: progress})
+                db.session.commit()
+
+            self.event_bus.push(
+                Event(
+                    type=EventType.TS_GENERATION_PROGRESS,
```

Comment:
This should not be an event specific to TS generation, but an event generic to all types of tasks: 
`TASK_PROGRESS`